nntrainer/layers/multi_head_attention_layer.cpp

```diff
@@ -73,10 +72,273 @@ enum AttentionParams {
   transposed_attention_output,
 };
 
-void MultiHeadAttentionLayer::finalizeCommon(InitLayerContext &context) {}
-
 void MultiHeadAttentionLayer::finalize(InitLayerContext &context) {
-  finalizeCommon(context);
+  if (context.getNumInputs() < 3 || context.getNumInputs() > 4) {
```

Comment:
because the finalize function is quite long, how about add some comments which explain more details about the tensors and weights.

Reply:
Done